### PR TITLE
code change for ticket 

### PR DESCRIPTION
**Pull request by ctopCanada**
code change for WET - 131 
Extract bubble from chat wizard and make it a separate plugin

### CODE REVIEW

**Comment by ctopCanada** [comment]
To fix those space and indentation problem , Is there any formattting tool we can use to comply with the format?

**Comment by GormFrank** [comment]
> To fix those space and indentation problem , Is there any formatting tool we can use to comply with the format?

Unfortunately we do not have such tool, you will have to look at the errors in the build and fix them manually in your file.

**Comment by ctopCanada** [comment]
I have fixed the js file format error, but the errors for _base.scss is really confusing. I couldn't figure it out at all. 
Is it possible that JSlint use JS coding conventions to check CSS files?

File: src/plugins/actionmng/actionmng.js [hunk 1]
```diff
@@ -1,9 +1,3 @@
-/**
- * @title WET-BOEW Action Manager
- * @overview API that coordinate actions with other wet-boew plugin
- * @license wet-boew.github.io/wet-boew/License-en.html / wet-boew.github.io/wet-boew/Licence-fr.html
- * @author @duboisp
- */
```
Comment (duboisp) [inline src/plugins/actionmng/actionmng.js:6]: Re-add the previous comment block, feel free to add you name in the "author" section

File: src/plugins/bubble/_base.scss [hunk 2]
```diff
@@ -0,0 +1,432 @@
+
+/*
+ * Web Experience Toolkit (WET) / Boîte à outils de l'expérience Web (BOEW)
+ * wet-boew.github.io/wet-boew/License-en.html / wet-boew.github.io/wet-boew/Licence-fr.html
+ * Chat wizard - Provisional CSS
```
Comment (duboisp) [inline src/plugins/bubble/_base.scss:5]: Change Chat wizard for "bubble"

File: src/plugins/bubble/_base.scss [hunk 3]
```diff
@@ -0,0 +1,432 @@
...
+
+.wb-bubble-bubble-wrap .bubble:focus {
+	border: 1px solid rgba(0, 0, 0, 0.5);
+	-webkit-box-shadow: 0 2px 3px rgba(0, 0, 0, 0.7);
+			box-shadow: 0 2px 3px rgba(0, 0, 0, 0.7); }
+
+.wb-bubble-btn-extrnl + .wb-bubble-bubble-wrap {
+	display: none !important; }
+
+/* Chat Wizard */
```
Comment (duboisp) [inline src/plugins/bubble/_base.scss:209]: Remove or replace this comment

Is the CSS bellow are only applicable to the bubble?

File: src/plugins/bubble/_base.scss [hunk 4]
```diff
@@ -0,0 +1,432 @@
...
+.wb-bubble-container .basic-link {
+	min-height: inherit; }
+
+.wb-bubble-mrgn {
+	margin-top: 80px; }
+
+.wb-bubble-container legend:focus {
+	outline: 1px dotted #666; }
+
+/* Chat wizard inline (full-screen) */
```
Comment (duboisp) [inline src/plugins/bubble/_base.scss:362]: Remove or replace this comment. 

Is the CSS bellow are only applicable to the bubble?

File: src/plugins/bubble/_base.scss [hunk 5]
```diff
@@ -0,0 +1,432 @@
...
+.wb-bubble-contained .conversation {
+	max-height: 70vh; }
+
+.wb-bubble-contained .minimize {
+	display: none; }
+
+.wb-bubble-contained .reset {
+	right: 0; }
+
+/* Chat wizard Media queries */
```
Comment (duboisp) [inline src/plugins/bubble/_base.scss:381]: Remove or replace this comment. 

Is the CSS bellow are only applicable to the bubble?

File: src/plugins/bubble/bubble.js [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+	/**
+	 * Initiate chat wizard bubble
+	 * @method initiateBubble
+	 * @param {jQuery DOM element} $selector Element which is the actual bubble
+	 */
+	initiateBubble = function( $selector ) {
+
+		var $footer = $( "#wb-info" ),
+			$link = $( selector + "-link", $selector );
+
```
Comment (duboisp) [inline src/plugins/bubble/bubble.js:31]: Add a check here to see if the page contain a actual footer, if no present, create a container after the "main" element.

File: src/plugins/bubble/bubble.js [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+			// Correct bubble positionning on load, on resize an on Y scroll if necessary
+			stickyUntilFooter( $selector );
+
+			$( window ).on( "resize scroll", function() {
+				stickyUntilFooter( $selector );
+			} );
+		}
+
+		// Open Chat from the notification message
+		$( ".notif", $selector ).on( "click", function() {
```
Comment (duboisp) [inline src/plugins/bubble/bubble.js:77]: Move this event outside the init function

File: src/plugins/bubble/bubble.js [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+			$link.click();
+		} );
+
+		// Close notification aside bubble
+		$( ".notif-close", $selector ).on( "click", function( event ) {
```
Comment (duboisp) [inline src/plugins/bubble/bubble.js:82]: Move this event outside the init function

File: src/plugins/bubble/bubble.js [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+			// Call my custom event
+			$elm.trigger( "name.of.your.event", settings );
+
+			// Identify that initialization has completed
+			wb.ready( $elm, componentName );
+		}
+	};
+
+// Add your plugin event handler  , register your custom event to the selector
+$document.on( "name.of.your.event", selector, function( event, data ) {
```
Comment (duboisp) [inline src/plugins/bubble/bubble.js:149]: This is just a sample code, you can re-use the logic here to move the event handler I asked to move outside the init function. 
Regarding this following code block, it need to be removed.

File: src/plugins/bubble/bubble.js [hunk 10]
```diff
@@ -0,0 +1,163 @@
...
+			list.insertBefore( li, list.childNodes[ 0 ] );
+
+
+			var $bubble = $( selector + "-bubble-wrap" );
+
+			// Initiate chat wizard bubble
+			initiateBubble( $bubble );
+
+			// Call my custom event
+			$elm.trigger( "name.of.your.event", settings );
```
Comment (duboisp) [inline src/plugins/bubble/bubble.js:141]: remove this line of code, it was just a sample and not used

File: src/plugins/bubble/bubble.js [hunk 11]
```diff
@@ -0,0 +1,163 @@
...
+
+			// Add link to disable WET plugins and polyfills
+			var list = document.getElementById( "wb-tphp" );
+			list.insertBefore( li, list.childNodes[ 0 ] );
+
+
+			var $bubble = $( selector + "-bubble-wrap" );
+
+			// Initiate chat wizard bubble
+			initiateBubble( $bubble );
```
Comment (duboisp) [inline src/plugins/bubble/bubble.js:138]: move the code contained in the function here or re-orgarnize it and provide meaningful name to function like "addingButton()" or "addStickySpacing" or...

**Review by duboisp — CHANGES_REQUESTED** [review]
We will need also a generic working example (with no chat but just opening a window), you can create the .hbs file into the bubble plugin folder, like other plugin.

The new action you added in the action manager need to be documented here like the other action: https://github.com/wet-boew/GCWeb/blob/master/src/plugins/doaction/doaction-doc-en.hbs#L249 
https://github.com/wet-boew/GCWeb/blob/master/src/plugins/doaction/doaction-doc-fr.hbs#L257

**Comment by duboisp** [comment]
@ctopCanada regarding the spacing, this repository have a .editorconfig which is doing it for you, you might probably need to configure your text editor in order to take advantage of it.

For the SASS it's the "grunt-sass-lint" nodejs module that perform that check. The error message in the Travis CI build told you what indentation is expected and the CSS property need to be alphabetical order too (aka property-sort-order )

File: src/plugins/actionmng/actionmng.js (hunk 1 above)
Comment (ctopCanada) [reply src/plugins/actionmng/actionmng.js:6]: completed as requested

File: src/plugins/bubble/bubble.js (hunk 11 above)
Comment (ctopCanada) [reply src/plugins/bubble/bubble.js:133]: sorry I didn't catch what you mean here. do you want to change the initiateBubble function name to addingButton?

**Comment by ctopCanada** [comment]
To squash all the commit into one, I did enter the command like this 

PS C:\temp\GCWeb> git reset --soft HEAD~40
PS C:\temp\GCWeb> git commit -m "code change for WET - 131 - Extract bubble from chat wizard and make it a separate plugin"
[master 7d53996] code change for WET - 131 - Extract bubble from chat wizard and make it a separate plugin
 8 files changed, 717 insertions(+), 5 deletions(-)
 create mode 100644 src/plugins/bubble/_base.scss
 create mode 100644 src/plugins/bubble/bubble-doc-en.hbs
 create mode 100644 src/plugins/bubble/bubble-doc-fr.hbs
 create mode 100644 src/plugins/bubble/bubble.js

**Pull request closed without merging**